### PR TITLE
Ctrl-click plant bags to change their appearance.

### PR DESCRIPTION
**Pull request by Allspice**
Basically copy-pasted the pill bottle color switcher. 14 variations. Sprites by realestestate.

:cl:
* rscadd: Ctrl-click plant bags to change their appearance.

### CODE REVIEW

**Comment by DrSnips** [comment]
Could we extend this to gadget, ore, etc bag too please? Cool stuff.
One other option is that you could make the straps an overlay and then use a color matrix to edit just that overlay so you could have any color.

**Comment by Allspice** [comment]
I can look into it later. Need to check through this list:

`code/modules/paperwork/clipboard.dm:/obj/item/weapon/storage/bag/clipboard
code/modules/paperwork/clipboard.dm:/obj/item/weapon/storage/bag/clipboard/New()
code/modules/paperwork/clipboard.dm:/obj/item/weapon/storage/bag/clipboard/update_icon()
code/modules/paperwork/clipboard.dm:/obj/item/weapon/storage/bag/clipboard/handle_item_insertion(obj/item/W as obj, prevent_warning = 0)
code/modules/paperwork/clipboard.dm:/obj/item/weapon/storage/bag/clipboard/remove_from_storage(obj/item/W as obj, atom/new_location, var/force = 0, var/refresh = 1)
code/modules/games/cards/wizard_cards.dm:/obj/item/weapon/storage/bag/wiz_cards
code/modules/games/cards/wizard_cards.dm:/obj/item/weapon/storage/bag/wiz_cards/full/New()
code/modules/games/cards/wizard_cards.dm:/obj/item/weapon/storage/bag/wiz_cards/attack_self(mob/user)
code/modules/games/cards/wizard_cards.dm:/obj/item/weapon/storage/bag/wiz_cards/show_to(mob/user as mob)
code/modules/games/cards/wizard_cards.dm:/obj/item/weapon/storage/bag/wiz_cards/frog/New()
code/game/machinery/kitchen/smartfridge.dm:/obj/machinery/smartfridge/proc/dump_bag(var/obj/item/weapon/storage/bag/B, var/mob/user)
code/game/machinery/machinery.dm:/obj/machinery/proc/exchange_parts(mob/user, obj/item/weapon/storage/bag/gadgets/part_replacer/W)
code/game/objects/items/robot/robot_items/robot_furnace.dm:/obj/item/weapon/storage/bag/ore/furnace
code/game/objects/items/robot/robot_items/robot_furnace.dm:/obj/item/weapon/storage/bag/ore/furnace/New()
code/game/objects/items/robot/robot_items/robot_furnace.dm:/obj/item/weapon/storage/bag/ore/furnace/update_icon()
code/game/objects/items/robot/robot_items/robot_furnace.dm:/obj/item/weapon/storage/bag/ore/furnace/proc/smelt_contents(var/mob/user)
code/game/objects/items/weapons/storage/RPED.dm:/obj/item/weapon/storage/bag/gadgets/part_replacer //Bag because disposals bin snowflake code is shit
code/game/objects/items/weapons/storage/RPED.dm:/obj/item/weapon/storage/bag/gadgets/part_replacer/discount_bluespace
code/game/objects/items/weapons/storage/RPED.dm:/obj/item/weapon/storage/bag/gadgets/part_replacer/proc/play_rped_sound()
code/game/objects/items/weapons/storage/RPED.dm:/obj/item/weapon/storage/bag/gadgets/part_replacer/attackby(var/obj/item/weapon/W, var/mob/user)
code/game/objects/items/weapons/storage/RPED.dm:/obj/item/weapon/storage/bag/gadgets/part_replacer/pre_loaded/New() //Comes preloaded with loads of parts for testing
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/trash
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/trash/update_icon()
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/plasticbag
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/plasticbag/suicide_act(mob/user)
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/ore
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/ore/auto
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/ore/auto/proc/auto_collect(var/turf/collect_loc)
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/ore/auto/proc/auto_fill(var/mob/holder)
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/ore/auto/proc/mob_moved(var/list/event_args, var/mob/holder)
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/ore/auto/dropped(mob/user)
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/plants
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/plants/CtrlClick()
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/plants/verb/change()
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/plants/portactor
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/food
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/food/update_icon()
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/food/menu1/New()
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/food/menu2/New()
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/food/borg
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/chem
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/sheetsnatcher
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/sheetsnatcher/New()
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/sheetsnatcher/can_be_inserted(obj/item/W as obj, stop_messages = FALSE)
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/sheetsnatcher/handle_item_insertion(obj/item/W as obj, prevent_warning = FALSE)
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/sheetsnatcher/orient2hud(mob/user as mob)
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/sheetsnatcher/quick_empty()
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/sheetsnatcher/remove_from_storage(obj/item/W as obj, atom/new_location)
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/sheetsnatcher/borg
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/gadgets
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/gadgets/mass_remove(atom/A)
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/money
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/money/treasure
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/money/treasure/New()
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/potion
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/potion/bundle
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/potion/bundle/New()
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/potion/lesser_bundle
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/potion/lesser_bundle/New()
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/potion/predicted_potion_bundle
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/potion/predicted_potion_bundle/New()
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/potion/lesser_predicted_potion_bundle
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/potion/lesser_predicted_potion_bundle/New()
code/game/objects/items/weapons/storage/bags.dm:/obj/item/weapon/storage/bag/ammo_pouch
code/datums/gamemode/role/rev.dm:/obj/item/weapon/storage/bag/ammo_pouch/rev
code/datums/gamemode/role/rev.dm:/obj/item/weapon/storage/bag/ammo_pouch/rev/New()
`

To see which are compatible and which look stupid with the stripe overlay

**Comment by 9percents** [comment]
can you post the images of the variations for clarity?

**Comment by Allspice** [comment]
plant bags

**Comment by jknpj** [comment]
Keep in mind that the borg protable seed extractor is a child of these. Make sure they don't get this.

File: code/game/objects/items/weapons/storage/bags.dm [hunk 1]
```diff
@@ -216,6 +236,10 @@ obj/item/weapon/storage/bag/plasticbag/quick_store(var/obj/item/I)
 		if(user.s_active)
 			user.s_active.show_to(user)
 
+
+/obj/item/weapon/storage/bag/plants/portactor/CtrlClick()
+	return
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/storage/bags.dm:241]: this is only okay because it's a borg item and therefore can't be dragged

**Pull request merged**